sopel/bot.py

```diff
@@ -484,6 +484,9 @@ def dispatch(self, pretrigger):
                     if (hasattr(func, 'intents') and
                             trigger.tags.get('intent') not in func.intents):
                         continue
+                    if (not (hasattr(func, 'echo') and func.echo is True) and
+                            trigger.nick.lower() == self.nick.lower()):
+                        continue
```

Comment:
I'd put this check first in the enclosing `for` loop, but the order isn't really an issue tbh.